.github/workflows/_build-and-test-locally.yml

```diff
@@ -28,6 +28,16 @@ on:
         required: false
         default: 'disabled'
         type: string
+      test-selection:
+        description: 'specification of selected test(s) to run'
```

Comment:
Let's use pytest's `-k` arg to select tests. It is more flexible than just passing a set of files. And we won't need to change `TEST_SELECTION` logic in `run-python-test-set` action

Reply:
The TEST_SELECTION logic there is still broken as of now (I mean -z condition after assigning the value):
```
        TEST_SELECTION="test_runner/${{ inputs.test_selection }}"
        ...
        if [ -z "$TEST_SELECTION" ]; then

```

Still, left this place as-is and turned test-selection into -k parameter.

Reply:
I wouldn't say it's broken, but the `if [ -z "$TEST_SELECTION" ]; then` is useless, you're right.
in the action input `test_selection` marked as required, so it will fail in the input is not set anyway

Reply:
That place in the "Run pytest" step is not changed in the current PR version.